packages/create-styles/src/cssCustomProperties/createRootStore.js

```diff
@@ -0,0 +1,97 @@
+import { getPropValue } from './getPropValue';
+import { hasVariable, isCustomProperty } from './utils';
+
+/**
+ * Stores CSS config variables that are expected to be added to :root.
+ * This is used for CSS variable fallback handling for IE 11.
+ */
+class RootStore {
+	state = {};
+
+	constructor(initialState = {}) {
+		this.setState(initialState);
+	}
+
+	/**
+	 * Retrieve a value from the state.
+	 * @param {string} key The key to retrieve.
+	 * @returns {string} The value.
+	 */
+	get = (key) => this.state[key];
+
+	/**
+	 * Retrieves the current state.
+	 * @returns {object} The state.
+	 */
+	getState = () => ({ ...this.state });
```

Comment:
Rather than spreading the state each time it's retrieved I wonder if we shouldn't just freeze it when we set it?

Reply:
Done!